source/fundamentals/connection.txt

```diff
@@ -0,0 +1,636 @@
+connect to a MongoDB deployment. It instructs the driver on how it should
+connect to MongoDB and how it should behave while connected. The following
+example shows each part of the connection URI:
+
+.. figure:: /includes/figures/connection_string_parts.png
+
+In this example, for the protocol, we use ``mongodb+srv`` which specifies the
+:manual:`DNS Seedlist Connection Format
+</reference/connection-string/#dns-seedlist-connection-format>`. This indicates
+that the hostname following it corresponds to the DNS SRV record of your
+MongoDB instance or deployment. If your instance or deployment does not have a
+DNS SRV record, use ``mongodb`` to specify the :manual:`Standard Connection
+Format </reference/connection-string/#standard-connection-string-format>`.
+
+.. note::
+
+   If your deployment is on MongoDB Atlas, follow the
+   :atlas:`Atlas driver connection guide <driver-connection?jmp=docs_driver_java>`
+   to retrieve your connection string.
+
+The next part of the connection string contains your username and password
+if you are using password-based authentication. Replace the value of ``user``
+with your username and ``pass`` with your password. If you are using an
+authentication mechanism that does not require a username and password, omit
+this part of the connection URI.
+
+The next part of the connection string specifies the hostname or IP address and
+port of your MongoDB instance. In the example below, we use ``sample-hostname``
+as the hostname and ``27017`` as the port. Replace these values to point to
+your MongoDB instance.
+
+The last part of the connection string contains connection and authentication
+options as parameters. In the example below, we set two connection options:
+``retryWrites=true`` and ``w=majority``. For more information on connection
+options, skip to the :ref:`connection options <connection-options>` section.
+
+The code below shows how you can use the sample connection URI in a client to
+connect to MongoDB.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/srv.java
+   :language: java
+
+.. _jndi:
+
+Java Naming and Directory Interface (JNDI)
+------------------------------------------
+
+MongoClientFactory includes a `JNDI <http://docs.oracle.com/javase/8/docs/technotes/guides/jndi/index.html>`__
+ObjectFactory implementation that returns ``MongoClient`` instances
+based on a :ref:`connection URI <connection-uri>`.
+
+.. _jndi-wildfly:
+
+Wildfly (formerly JBoss)
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+1. In a `Wildfly <http://wildfly.org/>`__ installation, create a new module
+   for MongoDB at `modules/system/layers/base/org/mongodb/main`. Copy the
+   ``mongo-java-driver.jar`` jar file into the module. Add the following
+   ``module.xml`` file into the module:
+
+   .. code-block:: xml
+
+      <module xmlns="urn:jboss:module:1.3" name="org.mongodb">
+         <resources>
+            <resource-root path="mongodb-driver-sync-4.1.0-rc0.jar"/>
```

Comment:
I'd like to propose moving code snippets that reference a specific version to an include with filename postfix "-with-version" so that we know to update it whenever a new version of the driver is released (or write a script to do that). How does that sound?

Reply:
Seems like a good idea to me.